debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/AbstractStreamingAdapter.java

```diff
@@ -13,6 +13,15 @@
  * @author Chris Cranford
  */
 public abstract class AbstractStreamingAdapter implements StreamingAdapter {
+
+    protected OracleConnectorConfig connectorConfig;
+    protected OracleConnection connection;
+
+    public void configure(OracleConnectorConfig connectorConfig, OracleConnection connection) {
```

Comment:
Can we rather make those constructor arguments, allowing to make fields final?

On that note, I just noticed that the two adapter types are instantiated reflectively. I don't think they should -- plain constructor calls are fine. I suppose the motivation to do so was that the XStream library might not be available. This shouldn't be a problem though, this code would only fail when actually trying to instantiate (and thus load) the `XStreamAdapter` class. I'd suggest to change this as part of this PR.

Reply:
It didn't hit me until after the build fails as to why `StreamingAdapter` implementations were loaded reflectively.  I did that because when we build on CI, we don't have access to the `xstream.jar` and we explicitly exclude all classes under the `oracle/xstream/` package so that we can effectively build the connector without that dependency.  Do we have a way in which we can source that jar from a private maven repository?

Reply:
That's interesting; even when _instantiating_ `XStreamAdapter` lazily, that very class itself must be compiled on CI. So I'm not quite sure yet how this currently actually is working?

Reply:
See the `oracle-ci` profile that is disabled by default but enabled by the Oracle CI workflow.  In that profile we customize the maven compiler plugin to explicitly exclude any classes under the `oracle/xstream` package.  We basically removed any specific reference during build time to any xstream class entirely so that the CI build would work, hence the lazy loading of the adapter to do so.